lib/tasks/rh_cloud_inventory.rake

```diff
@@ -27,6 +42,14 @@ namespace :rh_cloud_inventory do
         end
       end
     end
+    desc 'Upload generated inventory report to Red Hat cloud'
+    task upload: [:environment, 'dynflow:client'] do
+      base_folder = ENV['target'] || Dir.pwd
```

Comment:
I think `Dir.pwd` would point to the folder where the script resides in prod environment.
I would be better to set it to the default base folder:
```suggestion
      base_folder = ENV['target'] || ForemanInventoryUpload.generated_reports_folder
```

Reply:
I copied the line that we already have in `rh_cloud_inventory:report:generate`. But I do like your suggestion more, especially as it would fix one of issues reported in bugzilla (where running rake task on fresh Satellite fails, since directory structure does not exist yet).

Should I apply same change in `rh_cloud_inventory:report:generate`?

Reply:
I think defaulting to the `generated_reports_folder` could be problematic during report creation due to permissions - we can't be sure the user has permissions to write into foreman's folder. I suppose it would be better to default to creation in the current folder. 

We probably need to test it with `foreman-rake` to see if the folder is reported correctly though.

Reply:
It's `/usr/sbin/foreman-rake`, so it's already expected to be run by root/user with elevated permissions. Personally, I wouldn't worry about insufficient permissions here.

By the way, `Dir.pwd` isn't current working dir in shell - it seems to be changed to `/usr/share/foreman/` somewhere inside `foreman-rake` call:
```
# cd /tmp/
# export organization_id=1
# /usr/sbin/foreman-rake rh_cloud_inventory:report:generate
Must specify either portal_user or organization_id
I, [2020-11-18T06:34:00.043724 #25541]  INFO -- : Started generating hosts report in /tmp/d20201118-25541-suckmw
I, [2020-11-18T06:34:00.059385 #25541]  INFO -- : Adding slice 99a3216e-22bb-4a0c-8b54-934067e4a7ce
I, [2020-11-18T06:34:00.905352 #25541]  INFO -- : slice 99a3216e-22bb-4a0c-8b54-934067e4a7ce was created with 1 hosts
I, [2020-11-18T06:34:00.905479 #25541]  INFO -- : Report generation finished
I, [2020-11-18T06:34:00.905503 #25541]  INFO -- : Archiving generated report
I, [2020-11-18T06:34:00.909942 #25541]  INFO -- : tar: tar (child): /usr/share/foreman/report_for_1.tar.gz: Cannot open: Permission denied
tar (child): Error is not recoverable: exiting now
./
./metadata.json
./99a3216e-22bb-4a0c-8b54-934067e4a7ce.json

I, [2020-11-18T06:34:00.910009 #25541]  INFO -- : Tar command failed: pid 25562 exit 0
Successfully generated /usr/share/foreman/report_for_1.tar.gz for organization id 1
```